packages/lib-classifier/src/plugins/drawingTools/experimental/models/marks/FreehandLine/FreehandLine.js

```diff
@@ -3,278 +3,657 @@ import { FreehandLine as FreehandLineComponent } from '@plugins/drawingTools/com
+    splicePathSetup() {
+      // for purposes of undo/redo we use the action to set these properties
+      let action = self.drawActions.at(-1)
+      self.spliceClosePointIndex = action.pointIndexClose
+      self.spliceReverse = action.isReversed
+      self.spliceThroughBeginning = action.throughBeginning
+      self.spliceActive = true
+      self.setClosePoint(self.points[action.pointIndexClose])
     },
 
-    shortenPath() {
-      let lengthToRemove = 20
-      while (lengthToRemove--) {
-        self.pathX.pop()
-        self.pathY.pop()
+    spliceReset() {
+      self.splicePoints.length = 0
+      self.spliceActive = false
+      self.spliceReverse = false
+      self.spliceThroughBeginning = false
+      self.spliceDragPointIndex = null
+      self.spliceClosePointIndex = null
+    },
+
+    move() {
+      // do nothing... tool required but not used in this context
+    },
+
+    inactive() {
+      if (self.drawActions.length === 0) return // comes up in stories
+
+      while (['splice-drag-point', 'splice-close-point', 'splice-append'].indexOf(self.drawActions.at(-1).type) > -1) {
+        self.undo()
+        self.redoActionsClear()
       }
     },
 
-    splice(startIndex, endIndex) {
-      let firstPoint = Math.min(startIndex, endIndex)
-      let lastPoint = Math.max(startIndex, endIndex)
-      /*
-        Reverse the path, if necessary, so that the first point clicked
-        is always the draggable point.
-      */
-      if (firstPoint === endIndex) {
-        const lastIndex = self.points.length - 1
-        self.pathX.reverse()
-        self.pathY.reverse()
-        firstPoint = lastIndex - startIndex
-        lastPoint = lastIndex - endIndex
+    finish() {
+      if (self.points.length < self.minimumPoints) {
+        return self.tool.deleteMark(self)
       }
-      const deleteCount = lastPoint - firstPoint - 1
-      const clippedPoints = self.points.slice(firstPoint, lastPoint + 1)
-      self.clipPath = clippedPoints.map(({ x, y }) => ({ x, y }))
-      self.pathX.splice(firstPoint + 1, deleteCount)
-      self.pathY.splice(firstPoint + 1, deleteCount)
-      // Make the ends of the spliced section draggable.
-      self.dragPoint = self.points[firstPoint]
-      self.targetPoint = self.points[firstPoint + 1]
-    },
-
-    trim(startIndex, endIndex) {
-      let firstPoint = Math.min(startIndex, endIndex)
-      let lastPoint = Math.max(startIndex, endIndex)
-      /*
-        Reverse the path, if necessary, so that the first point clicked
-        is always at the end of the new line. The end of an open line
-        is the draggable point.
-      */
-      if (lastPoint === endIndex) {
-        const lastIndex = self.points.length - 1
-        self.pathX.reverse()
-        self.pathY.reverse()
-        lastPoint = lastIndex - startIndex
-        firstPoint = lastIndex - endIndex
+
+      if (self.isDragging) {
+        self.appendPathEnd()
+        self.finished = false
```

Comment:
This is causing a bug where you can't start drawing a new line after you've finished drawing the current line.

This should fix it:
```suggestion
        self.finished = true
```

Reply:
Changing this fixes keyboard focus too. I haven’t tested it thoroughly but it seems to work.